### ISSUE TITLE
--ignore-promoted-rules has no effect on copy_files

### ISSUE THREAD

**Issue by eponier** [ocaml/dune#10588]
## Actual Behavior
[The documentation](https://dune.readthedocs.io/en/stable/reference/dune/copy_files.html) of `copy_files` refers to [this page](https://dune.readthedocs.io/en/stable/reference/dune/rule.html#modes) about modes specifying that `--ignore-promoted-rules`  ignores rules with `(mode promote)`, but this not true for `copy_files`.

## Expected Behavior

I do not know if the proper fix is to change the code or change the documentation. In the first option, the code to change is [here](https://github.com/ocaml/dune/blob/38163a65a504f5fc4190a185d9562c33f43524ee/src/dune_rules/dune_file.ml#L73).

**Comment by emillon** [comment]
Good catch. The tricky part in fixing that is determining if we need to version the fix or not.
